test/tests/mail/model/MailListModelTest.ts

```diff
@@ -0,0 +1,481 @@
+		await model.loadInitial() // will have the first page loaded
+		const unloadedMail = makeMailSetElementId(1) // a mail that will be on the bottom of the list
+
+		// This mail is not loaded until we load a few more pages.
+		for (let loadedPageCount = 1; loadedPageCount < pages; loadedPageCount++) {
+			o(model.items.length).equals(PageSize * loadedPageCount)
+			const mail = model.getMailSetEntry(unloadedMail)
+			o(mail).equals(null)
+			await model.loadMore()
+		}
+
+		// Everything is loaded including that mail we wanted from before
+		o(model.items.length).equals(PageSize * pages)
+		const mail = model.getMailSetEntry(unloadedMail)
+		o(mail).notEquals(null)
+	})
+
+	o.test("loadAndSelect selects by mail id", async () => {
+		await setUpTestData(PageSize * 5, labels, false)
+		await model.loadInitial() // will have the first page loaded
+
+		// This mail is not loaded yet.
+		const unloadedMail = elementIdPart(makeMailId(1)) // a mail that will be on the bottom of the list
+		const mail = model.getMail(unloadedMail)
+		o(mail).equals(null)
+
+		// Should now be loaded
+		const loadedMail = await model.loadAndSelect(unloadedMail, () => false)
+		o(loadedMail).notEquals(null)
+		o(loadedMail).equals(model.getMail(unloadedMail))
+	})
+
+	o.spec("handleEntityUpdate", () => {
+		o.test("mailset update updates labels", async () => {
+			await setUpTestData(PageSize, [labels[0]], false)
+			await model.loadInitial()
+
+			// Overwrite one of the mails inside the list so it has both labels
+			const someIndex = 50 // a random number
+			const someMail: LoadedMail = {
+				...model._loadedMails()[someIndex],
+				labels: [labels[0], labels[1]],
+			}
+			someMail.mail.sets.push(labels[1]._id)
+			model._updateSingleMail(someMail)
+			o(model.getLabelsForMail(someMail.mail)[1]).deepEquals(labels[1])
+
+			// Change one of the labels (but not inside the mail we just updated)
+			labels[1] = {
+				...labels[1],
+				name: "Mint",
+				color: "#00FFAA",
+			}
+
+			o(model.getLabelsForMail(someMail.mail)[1]).notDeepEquals(labels[1])
+
+			const entityUpdateData = {
+				application: MailFolderTypeRef.app,
+				type: MailFolderTypeRef.type,
+				instanceListId: getListId(labels[1]),
+				instanceId: getElementId(labels[1]),
+				operation: OperationType.DELETE,
+			}
+
+			// delete and create have no effect
+			entityUpdateData.operation = OperationType.DELETE
+			await model.handleEntityUpdate(entityUpdateData)
+			o(model.getLabelsForMail(someMail.mail)[1]).notDeepEquals(labels[1])
+			entityUpdateData.operation = OperationType.CREATE
+			await model.handleEntityUpdate(entityUpdateData)
+			o(model.getLabelsForMail(someMail.mail)[1]).notDeepEquals(labels[1])
+
+			// but update updates
+			entityUpdateData.operation = OperationType.UPDATE
+			await model.handleEntityUpdate(entityUpdateData)
+			o(model.getLabelsForMail(someMail.mail)[1]).deepEquals(labels[1])
+
+			// verify getLabelsForMail call times
+			for (const mail of model.items) {
```

Comment:
this could be simpler with just checking that it was called twice for our specific email and once for some other email